### ISSUE TITLE
[Bug] Need error trap for unacceptable map names launching new data catalog

### ISSUE THREAD

**Issue by cmbarton** [OSGeo/grass#959]
**Describe the bug**
When wxGUI is launched, if there is a corrupted, deprecated, or other unacceptable map name somewhere in the GRASS data directory, an error is generated to the terminal and the data catalog does not initialize correctly (blank data window or lack of highlighting of current location/mapset, and "wait" cursor). Here is the error:

```Launching <wxpython> GUI in the background, please wait...
GRASS Arizona_SP_NAD83/PERMANENT:~ > Process Process-68:
Traceback (most recent call last):
  File "/Applications/GRASS-7.9.app/Contents/Resources/lib/python3.8/multiprocessing/process.py", line 315, in _bootstrap
    self.run()  
  File "/Applications/GRASS-7.9.app/Contents/Resources/lib/python3.8/multiprocessing/process.py", line 108, in run
    self._target(*self._args, **self._kwargs)  
  File "/Applications/GRASS-7.9.app/Contents/Resources/gui/wxpython/datacatalog/tree.py", line 169, in getLocationTree
    name, mapset = wholename.split('@')  
ValueError: too many values to unpack (expected 2)
```

@nilason made a patch with an added error trap that catches this, so that the data catalog initializes correctly. I made a slight modification to provide better info to the user to delete the map or change its name. It's probably better to just print the info to the terminal instead of g_message because 1) it has no negative impact on functionality once the bad map names are caught, and 2) the information will persist in the terminal can can be copied elsewhere, unlike a g_message window, to fix the problem. 

The patch is attached here. 

[tree_patch.txt](https://github.com/OSGeo/grass/files/5185120/tree_patch.txt)

**System description (please complete the following information):**

- Operating System: Noticed on Mac OSX, but probably affects any OS
- GRASS GIS version: 7.9 as of 7 Sep. 2020

**Comment by cmbarton** [comment]
I've added the fix as a PR.

**Comment by petrasovaa** [comment]
Could you describe how to reproduce this error?

**Comment by cmbarton** [comment]
In the test nc_08_grass7 test dataset, change the filename of /nc_spm_08_grass7/PERMANENT/cell/aspect to aspect@test. 

This will generate an error and keep the data catalog from completely initializing properly.

I ran into the error and only after @nilason added the new error trap in a patch did I discover some very old raster 3D files in a deprecated (I think) grid3 folder. No idea how these names got corrupted. There may be other things that create problem names. This catches at least some of them.

**Comment by wenzeslaus** [comment]
This might be a won't-fix because it is too localized. Yes, data catalog fails with corrupted database, but so does any other part of GRASS GIS.

A possible improvement here is to make the split there more robust and split only into two, never more, because that's how the mapname@mapsetname is defined.

If you want data catalog to deal with broken databases, that's a completely new issue/PR. One way how to address a part of that would be to check map names and mapset names for validity.

**Comment by cmbarton** [comment]
This section needs an error trap. It is the last of a series of read statements to create the listings in the data catalog, and the rest all have error traps. This particular error is likely rare. But it should not crash the entire data catalog on startup. Adding an error trap costs nothing and harms nothing. Mis-named maps elsewhere won't display or work in modules, but they don't crash major sections of the GUI (or shouldn't). 

What goes into the error trap is another matter. My suggestion is that the catalog just ignores the mis-named map (not at all a serious issue), but tells the user about it. This is what the other preceding traps do. I suggested a message in the terminal so that the user would not think it is serious and could more easily find the offending map names. Using g_message, as the other error traps do, is OK, just a little harder to use to track down problems in this case. 

An alternative like you suggest would be equally good. It would include the mis-named map in the data catalog, where the user could see it when it did not work. Either way, a valid file name but bad map name does not crash the whole thing and the user has a chance to discover problem maps.

**Comment by nilason** [comment]
I have a vague recollection of gui being buggy regarding selecting maps with mapset, I don't remember exactly what version (probably some 7.x) and maybe mac only. Maybe the bad names springs from that period.

I believe this issue is extremely rare and surprised this come up only now since the line L169 was introduced five years ago. Therefore I have no strong opinion on this.

A more forgiving alternative to try/except may be:

```py
            name, mapset, *rest = wholename.split('@')
            if rest:
                print(_("Warning: <{name}> ({type}) doesn't conform to GRASS GIS "
                      "standard \"mapname@mapset\". This may lead to unintended "
                      "consequences, you are strongly encouraged to rename this "
                      "map.").format(name=wholename, type=ltype))
                mapset = "{m}@{r}".format(m = mapset, r = "@".join(rest))
```
which transforms a bad map `themapname@PERMANENT@test` to map name `themapname` and mapset `PERMANENT@test` accompanying a warning.

Alternatively, ignoring the badly named map (similar to try/except):
```py
            name, mapset, *rest = wholename.split('@')
            if rest:
                print(_("Warning: <{name}> ({type}) doesn't conform to GRASS GIS "
                      "standard \"mapname@mapset\". This may lead to unintended "
                      "consequences, you are strongly encouraged to rename this "
                      "map.").format(name=wholename, type=ltype))
                continue
```

Any more (computationally) costly check (e.g. regex) would be unacceptable though.
